File: AirshipKit/AirshipKit/common/UASystemVersion.m
/* Copyright Urban Airship and Contributors */
#import "UAUtils+Internal.h"
#import "UASystemVersion+Internal.h"

@implementation UASystemVersion

+ (instancetype)systemVersion {
    return [[self alloc] init];
}

- (NSString *)currentSystemVersion {
    return UIDevice.currentDevice.systemVersion;
}

- (BOOL)isGreaterOrEqualToVersion:(NSString *)version {
    NSString *systemVersion = [self currentSystemVersion];

    NSComparisonResult result = [systemVersion compare:version options:NSNumericSearch];

    return result == NSOrderedSame || result == NSOrderedDescending;
}

@end

File: Airship/AirshipCore/Source/UASystemVersion.swift
// Copyright Airship and Contributors

import Foundation

/**
 * @note For internal use only. :nodoc:
 */
@objc
open class UASystemVersion : NSObject {

    @objc
    open var currentSystemVersion : String {
        get {
            return UIDevice.current.systemVersion
        }
    }

    @objc
    public func isGreaterOrEqual(_ version: String) -> Bool {
        let systemVersion = currentSystemVersion
        let result: ComparisonResult? = systemVersion.compare(version, options: .numeric, range: nil, locale: .current)
        return result == .orderedSame || result == .orderedDescending
    }
}
